File: ios-client/CursorMobile/CursorMobile/Models/Project.swift
import Foundation

struct Project: Codable, Identifiable, Hashable {
    let id: String
    let name: String
    let path: String
    let lastOpened: Date?
    
    enum CodingKeys: String, CodingKey {
        case id, name, path, lastOpened
    }
    
    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        id = try container.decode(String.self, forKey: .id)
        name = try container.decode(String.self, forKey: .name)
        path = try container.decode(String.self, forKey: .path)
        
        // Handle date parsing - API returns ISO8601 string
        if let dateString = try container.decodeIfPresent(String.self, forKey: .lastOpened) {
            let formatter = ISO8601DateFormatter()
            formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
            if let date = formatter.date(from: dateString) {
                lastOpened = date
            } else {
                // Try without fractional seconds
                formatter.formatOptions = [.withInternetDateTime]
                lastOpened = formatter.date(from: dateString)
            }
        } else {
            lastOpened = nil
        }
    }
    
    init(id: String, name: String, path: String, lastOpened: Date? = nil) {
        self.id = id
        self.name = name
        self.path = path
        self.lastOpened = lastOpened
    }
}

struct ProjectsResponse: Codable {
    let projects: [Project]
}

struct ProjectResponse: Codable {
    let project: Project
}

struct ProjectTree: Codable {
    let tree: [FileTreeItem]
}

struct FileTreeItem: Codable, Identifiable, Hashable {
    let name: String
    let path: String
    let isDirectory: Bool
    let children: [FileTreeItem]?
    
    var id: String { path }
}

struct CreateProjectRequest: Codable {
    let name: String
    let path: String?
    let template: String?
}

struct CreateProjectResponse: Codable {
    let success: Bool
    let project: NewProject?
    
    struct NewProject: Codable {
        let name: String
        let path: String
        let createdAt: String
    }
}

File: android-client/app/src/main/java/com/cursormobile/app/data/models/Project.kt
package com.cursormobile.app.data.models

import com.google.gson.annotations.SerializedName
import java.util.Date

data class Project(
    val id: String,
    val name: String,
    val path: String,
    val lastOpened: Date? = null
)

data class ProjectsResponse(
    val projects: List<Project>
)

data class ProjectResponse(
    val project: Project
)

data class ProjectTree(
    val tree: List<FileTreeItem>?
)

data class FileTreeItem(
    val name: String,
    val path: String,
    val type: String,
    val children: List<FileTreeItem>? = null,
    val size: Int? = null,
    @SerializedName("extension")
    val fileExtension: String? = null
) {
    val id: String get() = path
    val isDirectory: Boolean get() = type == "directory"
}

data class CreateProjectRequest(
    val name: String,
    val path: String? = null,
    val template: String? = null
)

data class CreateProjectResponse(
    val success: Boolean,
    val project: NewProject? = null
) {
    data class NewProject(
        val name: String,
        val path: String,
        val createdAt: String
    )
}
